Sources/Safetensors/Enums/DType.swift

```diff
@@ -0,0 +1,20 @@
+//
+// Created by Tomasz Stachowiak on 1.10.2024.
+//
+
+import Foundation
+import CoreML
+
+
+public enum DType: String, Codable {
+    case float64 = "F64"
+    case float32 = "F32"
+    case float16 = "F16"
+    case int32 = "I32"
+    case uint32 = "U32"
+    case int16 = "I16"
+    case uint16 = "U16"
+    case int8 = "I8"
+    case uint8 = "U8"
+    case bool = "BOOL"
+}
```

Comment:
the original lib implements more dtypes, see here https://github.com/huggingface/safetensors/blob/5db3b92c76ba293a0715b916c16b113c0b3551e9/safetensors/src/tensor.rs#L656

the problem I see with this approach is that even more types can be added in future. Keeping as a string allows us to parse it without a risk that a future addition will break it. We are validating it anyway when materializing to `MLTensor` of `MLMultiArray`

Reply:
we have to check support for the new string types anyway so at least we have clearly defined place where to add it

Reply:
let consider a scenario where a new type was added but we don't support it yet. The example file contains bunch of tensors, some of them we support, some of them we don't (they are not specified in `DType` enum). I might be wrong but using your approach user won't be able even to open the file, using the old approach user will be able to open the file and materialize the tensors we have the support for

Reply:
do you see value in being able to open the file but not able to materialise it?

Reply:
a file can contain tensors data of multiple type, so yes, I see the value of being able to open the file and materialize the tensors we have the support for